Sources/Features/Breadcrumb/QueueFile.swift

```diff
@@ -1,5 +1,15 @@
 import Foundation
 
+struct BreadcrumbRecord {
```

Comment:
I think it should be moved to a separated file. 
You implemented queue to be generic. However, putting BreadcrumbRecord definition in the Queue implementation sounds weird.

Reply:
moving it